geode-core/src/main/java/org/apache/geode/internal/cache/BucketAdvisor.java

```diff
@@ -1605,21 +1605,34 @@ protected Profile instantiateProfile(InternalDistributedMember memberId, int ver
       InternalCache cache = getProxyBucketRegion().getCache();
       List<CacheServer> servers = cache.getCacheServers();
 
-      HashSet<BucketServerLocation66> serverLocations = new HashSet<>();
-      for (CacheServer cacheServer : servers) {
-        CacheServerImpl server = (CacheServerImpl) cacheServer;
+      HashSet<BucketServerLocation66> serverLocations = getBucketServerLocations(version, servers);
+      if (serverLocations.size() > 0) {
+        return new ServerBucketProfile(memberId, version, getBucket(), serverLocations);
+      }
+    }
+    return new BucketProfile(memberId, version, getBucket());
+  }
+
+  protected HashSet<BucketServerLocation66> getBucketServerLocations(int version,
+      List<CacheServer> servers) {
+    HashSet<BucketServerLocation66> serverLocations = new HashSet<>();
+    for (CacheServer cacheServer : servers) {
+      CacheServerImpl server = (CacheServerImpl) cacheServer;
+      try {
         if (server.isRunning() && (server.getExternalAddress() != null)) {
           BucketServerLocation66 location = new BucketServerLocation66(getBucket().getId(),
               server.getPort(), server.getExternalAddress(), getBucket().isPrimary(),
               Integer.valueOf(version).byteValue(), server.getCombinedGroups());
           serverLocations.add(location);
         }
-      }
-      if (serverLocations.size() > 0) {
-        return new ServerBucketProfile(memberId, version, getBucket(), serverLocations);
+      } catch (IllegalStateException e) {
+        logger.warn("Cache server's bind address is not available because it may not have started");
```

Comment:
I don't see any reason to log a warning here. Also, I think we should only see this exception if the server was stopped right after we called isRunning(). But we should not warn them about having stopped a server. I also don't see a need for the debug logging.
I do think you should make sure this is the IllegalStateException thrown by getExternalAddress. Check it for the expected message. If it is some other IllegalStateException then rethrow it. Put the message string in a constant that this code can reference.

Reply:
done